scripts/hal_alignment/hal_gene_liftover.py

```diff
@@ -49,17 +50,43 @@ class SimpleRegion(NamedTuple):
     strand: str
 
 
+def load_chrom_sizes(in_hal_file: Union[Path, str], genome_name: str) -> Dict[str, int]:
+    """Load chromosome sizes from an input HAL file.
+
+    Args:
+        in_hal_file: Input HAL file.
+
+    Returns:
+        Dictionary mapping chromosome names to their lengths.
+
+    """
+    cmd = ['halStats', '--chromSizes', genome_name, in_hal_file]
+    process = run(cmd, stdout=PIPE, check=True)
+    text = process.stdout.decode('ascii')
+
+    chrom_sizes = dict()
+    with io.StringIO(text) as stream:
+        for line in stream:
+            chrom, chrom_size = line.rstrip().split('\t')
+            chrom_sizes[chrom] = int(chrom_size)
```

Comment:
`subprocess.run` added the `text` and `capture_output` options in `3.7` which can handle the process of capturing and decoding the stdout for you. Using `StringIO` for the purposes of iterating the lines in a string seems a little overkill compared to `splitlines`

```  suggestion
    process = run(cmd, check=True, capture_output=True, text=True,encoding='ascii')

    chrom_sizes = dict()
    for line in process.stdout.splitlines():
        chrom, chrom_size = line.rstrip().split('\t')
        chrom_sizes[chrom] = int(chrom_size)
```